From 093d49af2773bf017cf97ff2054a405b900b84ce Mon Sep 17 00:00:00 2001
Message-Id: <093d49af2773bf017cf97ff2054a405b900b84ce@dist-git>
From: Martin Kletzander <mkletzan@redhat.com>
Date: Fri, 3 Apr 2015 18:50:11 +0200
Subject: [PATCH] qemu: Fix possible crash

https://bugzilla.redhat.com/show_bug.cgi?id=1198497

Just a missing 'goto cleanup;' after reporting an error.

RHEL-only; upstream was totally reworked.
Signed-off-by: Martin Kletzander <mkletzan@redhat.com>

Signed-off-by: Martin Kletzander <mkletzan@redhat.com>
Signed-off-by: Jiri Denemark <jdenemar@redhat.com>
---
 src/qemu/qemu_driver.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/qemu/qemu_driver.c b/src/qemu/qemu_driver.c
index 3f99795..371dcbe 100644
--- a/src/qemu/qemu_driver.c
+++ b/src/qemu/qemu_driver.c
@@ -8370,6 +8370,7 @@ qemuDomainSetNumaParamsLive(virDomainObjPtr vm,
             virReportSystemError(-rc, _("Unable to access vcpu cgroup for "
                                         "%s (vcpu: %zu)"),
                                  vm->def->name, i);
+            goto cleanup;
         }
         if ((rc = virCgroupSetCpusetMems(cgroup_temp, nodeset_str)) < 0) {
             virReportSystemError(-rc, _("Failed to set cpuset.mems to %s"),
-- 
2.3.5

